agent/engine/docker_task_engine.go

```diff
@@ -771,6 +771,11 @@ func (engine *DockerTaskEngine) pullAndUpdateContainerReference(task *api.Task,
 		defer container.SetRegistryAuthCredentials(credentials.IAMRoleCredentials{})
 	}
 
+	//if using asm auth, tag the asm authData with taskID
+	if container.IsASMImagePullAuth() {
```

Comment:
It looks like these two methods are always used together, can we combine them into one?

Reply:
hm good point